src/bluez/adapter/peripheral.rs

```diff
@@ -310,6 +315,26 @@ impl Peripheral {
                 .collect();
         }
 
+        if let Some(service_data) = args.service_data() {
+            properties.service_data = service_data
+                .into_iter()
+                .filter_map(|(service, data)| {
+                    let service: UUID = service.parse().unwrap();
+                    if let Some(data) = cast::<Vec<u8>>(&data.0) {
+                        self.adapter.emit(CentralEvent::ServiceAdvertisement {
+                            address: self.address(),
+                            service: service,
```

Comment:
Nit: You can just write `service`, as the field name and variable name are the same.